### ISSUE TITLE
Does the environment monitor node work?

### ISSUE THREAD

**Issue by mpowelson** [tesseract-robotics/tesseract_ros2#14]
I'm trying to use the environment monitor node to test tesseract_rviz. It doesn't seem like it is publishing the state like I would expect. Have we used it anywhere or was this a partial port that just built? I notice there is still a todo in the cmake saying the @schornakj should port it.

**Comment by schornakj** [comment]
Yes, we use this node in some of our projects, though it's definitely possible that the aspects of the node that were important to those projects are different form the ones you need to use for `tesseract_rviz`.

**Comment by afrixs** [comment]
@mpowelson are you experiencing the same behavior like me (see image) when running `ros2 launch tesseract_ros_examples pick_and_place_example.launch.xml`?
`ros2 topic echo /tesseract_ros_examples/tesseract_published_environment` shows there is a -1.57 rotation of `iiwa_joint_a4` but the arm displayed in rviz looks like all joints have zero positions... also moving manipulation sliders affects only the "interactive" markers (which are not interactive - I cannot drag them), but arm doesn't move.

![image](https://user-images.githubusercontent.com/6342166/226946323-ddf25191-0764-4b3d-9c8e-bfef7829e7bb.png)

According to logs and little axes in RViz, trajectory has been planned and executed successfully but I cannot see the arm following it. There are Qt errors when starting RViz:
```
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'QList<QPersistentModelIndex>'
[rviz2-2] (Make sure 'QList<QPersistentModelIndex>' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'Qt::Orientation'
[rviz2-2] (Make sure 'Qt::Orientation' is registered using qRegisterMetaType().)
[rviz2-2] QObject::connect: Cannot queue arguments of type 'tesseract_environment::Environment'
[rviz2-2] (Make sure 'tesseract_environment::Environment' is registered using qRegisterMetaType().)
```

They seem to be related to the issue, but I can't find an elegant way to get rid of them (trails point towards `tesseract_qt`, but that's where I stopped because the change seems to be quite big and invasive). @schornakj do you have any clues on what I should do to prevent the errors and buggy behavior? Or is this behavior correct?
Note: I just built `tesseract_ros2` along with dependencies and ran `ros2 launch tesseract_ros_examples pick_and_place_example.launch.xml` because I could not find any closer guide on how to run the tutorials (should gazebo be launched as well or something?) and what exactly I should see...

OS: Ubuntu 22.04.1 LTS
ROS Humble
`ros-humble-rviz-common` version: 11.2.5-1jammy.20230302.200806
Qt version: 6.2.4

**Comment by rjoomen** [comment]
@marrts and me are working on parity with tesseract_ros 0.15, see https://github.com/marrts/tesseract_ros2/tree/0_15-parity and PR.

Now the interactive markers don't show anymore, but at least the robot model moves when using the manipulation sliders. The visualization still starts with the robot zeroed, but changing the display mode of the workbench from Monitor to URDF and back again makes it show the correct pose. Most Qt errors are still present, but the one regarding `tesseract_environment::Environment` is gone.

**Comment by afrixs** [comment]
@rjoomen thank you, I have finally made the examples working!
I must note though, that several errors needed to be fixed across tesseract to make things running without crashes (one very rare crash remained but I think it's currently low-priority, hard to reproduce, and maybe gets fixed by fixing the `QObject::connect` errors).
The fixes are in following PRs: tesseract-robotics/tesseract_planning#298 tesseract-robotics/tesseract_qt#32 marrts/tesseract_ros2#6

I've got one more question: Currently the planning succeeds and planned trajectories can be seen and played in Trajectory tab in Rviz one by one, but it would be nice to see the whole movement at once right away after planning. I think it's most desired in the `online_planning_example` where tons of trajectories are being created (flooding RAM very quickly) but no motion is visible until a trajectory is clicked and played and thus the "online" experience is missing... Are there plans for the automatic planned motion execution visualization to be added? (Or is it possible to visualize it if it's already been implemented somewhere?)

This is what I currently see before hitting the green play button
![image](https://user-images.githubusercontent.com/6342166/228312063-72158241-38e8-473c-9b01-ed7b451a4509.png)
And after hitting play button (with animation from original state)
![image](https://user-images.githubusercontent.com/6342166/228312132-73d83ef7-ce80-408c-9128-f55b662f3ec7.png)

And this kind of image is in README.md (and it's what I'd like to achieve)
![Online Planning Example](https://github.com/tesseract-robotics/tesseract_ros2/raw/master/gh_pages/_static/examples/online_planning_example.gif)
